packages/opentelemetry-node/src/instrumentation/PluginLoader.ts

```diff
@@ -83,6 +83,18 @@ export class PluginLoader {
         return this;
       }
 
+      const alreadyRequiredModules = Object.keys(require.cache);
+      const requiredModulesToHook = modulesToHook.filter(name =>
+        alreadyRequiredModules.some(cached => cached.includes(name))
```

Comment:
Could this not be a problem where some already required module includes the name of the module we are requiring but is not the module we are requiring? For instance, if some fictional module `http-logging-helper` is previously required, will `require("http")` trip this?

Reply:
> where some already required module includes the name of the module we are requiring but is not the module we are requiring

If `http-logging-helper` require `http`, it will be present in `require.cache`, it's a flat structure. So if i understood correctly it should not be a problem

Reply:
But if `http-logging-helper` exists in the `alreadyRequiredModules`, then `cached.includes('http')` will evaluate to true.

Reply:
Right, i must update that to a regex to take that into account. can you request changes on the PR to be sure that we don't merge it ?

Reply:
Updated to use a regex !